waypointfollowing.yml

```diff
@@ -0,0 +1,170 @@
+version: '2'
+
+services:
+######################################################################################
+#
+# Microservices for RHINO (Volvo FH16).
+#
+######################################################################################
+
+    # Relay and downsample Envelopes from CID 111 (local CID session on Brick) to CID 110 (for opendlv-vehicle-view).
+    relay-downsample-101-100:
+        container_name: relay-downsample-101-100
+        image: chrberger/cluon-relay-multi:v0.0.4
+        cpuset: "0"
+        restart: always
+        network_mode: "host"
+        cap_add:
+        - SYS_NICE
+        entrypoint: ""
+        command: "nice -n -7 /usr/bin/cluon-relay --cid-from=101 --cid-to=100 --downsample=19:10,49:1,1046:10,51:1"
+     
+    # Relay cluon.data.RecorderCommand.
+    relay-recordercommand-actuationrequest-100-101:
+        container_name: relay-recordercommand-actuationrequest-100-101
+        image: chrberger/cluon-relay-multi:v0.0.4
+        cpuset: "0"
+        restart: always
+        network_mode: "host"
+        cap_add:
+        - SYS_NICE
+        entrypoint: ""
+        command: "nice -n -7 /usr/bin/cluon-relay --cid-from=100 --cid-to=101 --keep=11,160"
+
+    ###########################################################################
+    # Dump all containers from CID 101 on request.
+    record-101:
+        container_name: record-101
+        image: chrberger/cluon-record-multi:v0.0.1
+        cpuset: "0"
+        restart: always
+        network_mode: "host"
+        volumes:
+        - $HOME/recordings:/recordings
+        working_dir: /recordings
+        command: "--cid=101 --remote --recsuffix=-all"
+    
+    ###########################################################################
+    # OxTS GPS (10.40.40.56)
+    dev-gps-ncom:
+        container_name: dev-gps-ncom
+        image: chalmersrevere/opendlv-device-gps-ncom-multi:v0.0.17
+        cpuset: "1"
+        restart: always
+        network_mode: "host"
+        cap_add:
+        - SYS_NICE
+        entrypoint: ""
+        command: "nice -n -5 /usr/bin/opendlv-device-gps-ncom --ncom_ip=0.0.0.0 --ncom_port=3000 --cid=101 --id=0"
+    
+    ############################################################################
+    # Velodyne VLP32c Lidar
+    dev-lidar-vlp32c:
+        container_name: dev-lidar-vlp32c
+        image: chalmersrevere/opendlv-device-lidar-vlp32c-multi:v0.0.3
+        cpuset: "1"
+        restart: always
+        network_mode: "host"
+        cap_add:
+        - SYS_NICE
+        entrypoint: ""
+        command: "nice -n -5 /usr/bin/opendlv-device-lidar-vlp32c --vlp32c_ip=10.40.40.1 --vlp32c_port=2368  --cid=101 --id=0"
+
+    ###########################################################################
+    # Axis camera including recording functionality with remote start/stop.
+    dev-camera-axis-out:
```

Comment:
We need to plan a systematic experiment here: mjpegoverhttp vs the h264/RTP version to compare quality, jitter, latency and robustness to network congestion to decide what microservice we shall use in the future (on Eagle, we changed from mjpeg to h264/RTP). Could this be something for the summer workers?

Reply:
Yes, but they would need to be guided and you will need to elaborate on how this can be tested systematically. The summer interns are working till mid Aug i think